editor-components/stats/stats5/stats5.tsx

```diff
@@ -58,42 +62,168 @@ class Stats5Page extends BaseStats {
+    }
+    this.startAnimation();
+  }
+
+  startAnimation() {
+    this.intervalId = setInterval(() => {
+      let allCompleted = true;
+      this.castToObject<CardData[]>("cardContent").forEach(
+        (statsData: CardData, index: number) => {
+          let statNumber = this.getComponentState(`number-${index}`);
+          if (statNumber !== statsData.CardTitle) {
+            allCompleted = false;
+            this.setComponentState(
+              `number-${index}`,
+              Math.min(
+                statsData.CardTitle,
+                statNumber +
+                  Math.ceil(
+                    statsData.CardTitle /
+                      Math.round(this.getPropValue("animation-duration") / 30)
+                  )
+              ) || 0
+            );
+          }
+        }
+      );
+
+      if (allCompleted && this.intervalId) {
+        clearInterval(this.intervalId);
+        this.intervalId = null;
+      }
+    }, 30);
+  }
+
+  componentWillUnmount() {
+    if (this.intervalId) {
+      clearInterval(this.intervalId);
+    }
   }
 
   getName(): string {
     return "Stats 5";
   }
 
+  getCardClasses(index: number, itemCountInRow: number) {
+    const totalCards = this.castToObject<CardData[]>("cardContent").length;
+    const isLastInRow =
+      (index + 1) % itemCountInRow === 0 || index === totalCards - 1;
+
+    return !isLastInRow ? this.decorateCSS("stick") : "";
+  }
+
   render() {
+    const itemCountInRow = this.getPropValue("itemCountInRow");
+    const hasCard = this.castToObject<CardData[]>("cardContent").length > 0;
+
     return (
       <div className={this.decorateCSS("container")}>
         <div className={this.decorateCSS("max-content")}>
-          <div className={this.decorateCSS("bottom-child")}>
-            {this.castToObject<ICard[]>("card-content").map(
-              (cardData: any, indexCard: number) => (
-                <div key={indexCard} className={this.decorateCSS("card")}>
-                  <h4 className={this.decorateCSS("card-data-title")}>{cardData.title}</h4>
-                  <p className={this.decorateCSS("card-data-description")}>{cardData.description}</p>
-                </div>
-              )
-            )}
-          </div>
+          {hasCard && (
+            <div className={this.decorateCSS("bottom-child")}>
+              {this.castToObject<CardData[]>("cardContent").map(
+                (data: CardData, index: number) => (
+                  <div
+                    key={index}
+                    className={`${this.decorateCSS(
+                      "card"
+                    )} ${this.getCardClasses(index, itemCountInRow)}`}
+                    style={{
+                      width: 90 / itemCountInRow + "%",
+                    }}
+                  >
+                    <h4 className={this.decorateCSS("card-data-title")}>
+                      {this.getComponentState(`number-${index}`)}
+                    </h4>
```

Comment:
don't render h4 if {this.getComponentState(`number-${index}`)} is empty
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)